pkg/annotations/annotations.go

```diff
@@ -0,0 +1,55 @@
+package annotations
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// Represents a single operation in a JSON Patch request body
+// Examples: {"op":"add","path":"/metadata/annotations/foo","value":"bar"}
+//           {"op":"remove","path":"/metadata/annotations/baz"}
+type patchOperation map[string]string
```

Comment:
I feel like what you are trying to do here is better achieved by using a struct. I dont see any benefit in defining the `patchOperation` in this way.

Reply:
The reasoning behind keeping it more fluid was that `remove` operations do not have the `value` field.

Do you think creating 2 separate structs, one for `add` operations and one for `remove` would be more clear?

Reply:
I like the 2 struct idea, unless @sebsoto has another suggestion

Reply:
maybe you could reuse the patching we do here using [struct](https://github.com/openshift/windows-machine-config-operator/blob/892d1850ad645f57097c670a2560c397f3f9a244/pkg/metrics/metrics.go#L72) in the metrics pkg.

Reply:
+1 to @mansikulkarni96 comment. We should have one method for of patching in our code. 

Moreover, standalone functions like these need unit tests.

Reply:
I am not clear on we could consolidate things to one Patch method. The patching done the metrics pkg is a different operation (`replace`) on a different k8s object (`Endpoints` vs. `Nodes`), so the patch body will be different. At minimum there would have to be separate method to process the each type of object.